src/app/tests/TestNullable.cpp

```diff
@@ -0,0 +1,307 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <array>
+#include <inttypes.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <app/data-model/Nullable.h>
+#include <lib/support/Span.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+
+using namespace chip;
+using namespace chip::app::DataModel;
+
+struct Count
+{
+    Count(int i) : m(i) { ++created; }
+    ~Count() { ++destroyed; }
+
+    Count(const Count & o) : m(o.m) { ++created; }
+    Count & operator=(const Count &) = default;
+
+    Count(Count && o) : m(o.m) { ++created; }
```

Comment:
is this safe? If I move something, I expect its destructor to not be called. Is this intentional that if I do move semantics only created increased but not destroyed?